### PR TITLE
Add `repo` option to narrow down token scope

### LINKED ISSUE #174: Add permissions input to narrow down permissions on a given token request.
When a token is generated, if we don't pass a permissions input during the request, that token comes with all the allowed permissions available in the App.

Sometimes we want to narrow down those permissions in a given token generation. 

https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#create-an-installation-access-token-for-an-app

### PR DESCRIPTION
**Pull request by kpassapk**
Adds a "repo" option which can be used to restrict the token scope to only a subset of the repositories that the installed application has been granted. If the option is not given, a token with the maximum scope is returned.

```
> github-app-installation-token --appId x --installationId y --privateKeyLocation z --repo repo1 --repo repo2
```

It relates to #174

### CODE REVIEW

File: src/bin/cli.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ import { getTokenCommand } from '../commands';
 
 program.name(packageJSON.name).description(packageJSON.description).version(packageJSON.version);
 
+function collectRepos(value: string, previous: string[]) {
+  return previous.concat([value]);
+}
+
```
Comment (gagoar) [inline src/bin/cli.ts:12]:
```suggestion
const collectRepos = (value: string, previous: string[]) => [...previous, value];
```

File: src/bin/cli.ts [hunk 2]
```diff
@@ -25,6 +29,7 @@ program
     '--baseUrl <baseUrl>',
     'Change the base url for github request. For example if used on a Github Enterprise Server instance.'
   )
+  .option('-r, --repo <name>', 'Allow access to a single repository', collectRepos, [])
```
Comment (gagoar) [inline src/bin/cli.ts:32]:
```suggestion
  .option('-r, --repo <name>', 'Allow access to a single repository', (value: string, previous: string[]) => [...previous, value], [])
```

Comment (gagoar) [reply src/bin/cli.ts:32]: This is another option. Given that is a one liner, written this way can be easier to just leave it there in the option.

File: src/bin/cli.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kpassapk) [reply src/bin/cli.ts]: Let's go for the inline version.

File: package.json [hunk 3]
```diff
@@ -114,7 +114,7 @@
   "devDependencies": {
     "@types/debug": "4.1.7",
     "@types/jest": "26.0.24",
-    "@types/node": "16.11.43",
+    "@types/node": "^16.11.43",
```
Comment (gagoar) [inline package.json:117]: please don't change the semver direction of this package.

**Comment by gagoar** [comment]
I've seen what DeepSource is telling us here, and I'm a bit more than against the suggestions. It recommends things without context, assuming console.log is triggered in the browser. Clearly DeepSource needs configuration that goes the ROI at this point. So I've disabled.

**Comment by gagoar** [comment]
There are 2 pending things now:
1) rever the change in package.json
2) the title is wrong. this doesn't close the ticket just adds to it. The ticket will need the permissions field also populated that can happen in another PR.

File: package.json (hunk 3 above)
Comment (kpassapk) [reply package.json]: Reverted

**Comment by kpassapk** [comment]
Done

**Review by gagoar — APPROVED** [review]

**Pull request merged**